File: Single Element in a Sorted Array.cpp
class Solution {
public:
    int singleNonDuplicate(vector<int>& nums) {
        int ans=0;
        int i=0;
        if(nums.size()==1)
        {
            return nums[0];
        }
        while(i<nums.size()-1)
        {
            if(nums[i]!=nums[i+1])
            {
                ans=nums[i];
                break;
            }
            else
            {
                i+=2;
            }
        }
        if(ans==0)
        {
            if(nums[nums.size()-1]!=nums[nums.size()-2] && nums[nums.size()-3]==nums[nums.size()-2] )
            {
                ans = nums[nums.size()-1];
            }
        }
        return ans;
        
    }
};

File: Single Element in a Sorted Array.java
class Solution {
    public int singleNonDuplicate(int[] nums) {
        int uni=Integer.MIN_VALUE;
        int n=nums.length;
        if(n==1){
            return nums[0];
        }
        for(int i=0;i<n;i++){
            if(i==n-1){
                uni=nums[i];
                break;
            }
            if(nums[i]==nums[i+1]){
                i++;
            }else{
                uni=nums[i];
                break;
            }
        }
        return uni;

    }
}
